### DISCUSSION TITLE
Running tests

### DISCUSSION THREAD

**Discussion by AGeissler** [NTNU-IHB/PythonFMU#197] [Q&A] [answered]
Hi there, a short question: how do you run the tests included in the package? Installation seems fine, the build of pythonfmu-export also worked flawlessly, it seems (on MacOS 13.6, Apple Silicone); creating an FMU via  pythonfmu build -f pythonslave.py also seemed to work, however, in the FMU subfolder "binaries" there is only linux64 and win64 - no idea, if this is an (additional) issue. However, running the test_*.py scripts does not seem to lead to any actual running of any FMUs. I was unable to find an actual "tester". Thanks for any ideas. Best, Achim.

**Comment by markaren** [comment, marked as answer]
The pythonfmu distribution from pip/conda will only include binaries for x64 Windows and Linux. However, if you build pythonfmu-export it should generate binaries for OSX in some folder that can be copied into the FMU. If the package actually work on OSX, however, I have no idea. As for tests, they should run when using pytest?

**Reply by AGeissler** [reply]
Thanks! Yes, I was actually unaware how pytest works (never used it before). Now the tests are being run, it seems, but there are issues (I will need to analyse these and see if I can make heads and tails of them). 

I succeeded in building "pythonfmu-export", leading to ./resources/binaries/darwin64/libpythonfmu-export.dylib, yes. Is this the file that should replace .so / .dll output in an FMU? Simply rename and copy there? Thanks for any additional hints and ideas. Best, Achim

**Reply by markaren** [reply]
Just copy `darwin64/libpythonfmu-export.dylib` into the binaries folder into the FMU. The FMU should then in theory work on linux, windows and osx (all 64bit).

**Reply by AGeissler** [reply]
O.K., this seems to work. After amending tests/slaves/pythonslave.py by 
-        self.register_variable(Integer("intOut", causality=Fmi2Causality.output))
+        self.register_variable(Integer("intOut", causality=Fmi2Causality.output, variability=Fmi2Variability.constant))
as well as for 'booleanVariable' and 'stringVariable' the tests had no more errors. Producing the .dylib was simply done by running build_unix.sh. If the "variability=constant" makes any sense for actual usage, I have no idea. But if yes, it may be a fix for the repo. Also, darwin64 support could be added for install, maybe - it seems to work if the tests are exhaustive. Thanks again, best, Achim.
